app/post.go

```diff
@@ -1116,7 +1116,8 @@ func (a *App) GetFileInfosForPostWithMigration(postId string) ([]*model.FileInfo
 		post := result.Data.(*model.Post)
 
 		if len(post.Filenames) > 0 {
-			a.Srv.Store.FileInfo().InvalidateFileInfosForPostCache(postId)
+			a.Srv.Store.FileInfo().InvalidateFileInfosForPostCache(postId, false)
+			a.Srv.Store.FileInfo().InvalidateFileInfosForPostCache(postId, true)
```

Comment:
Why do we call this twice? Should we always just invalidate for deleted as well?

Also, shouldn't we remove from the FileInfo cache when `DeleteForPost` is called?

Reply:
I guess so, but wasn't there before, so I'm only migrating the existing implementation in the new cache layer, but I'm ok adding that to other places, like `PermanentDelete`, `PermanentDeleteBatch`, `PermanentDeleteByUser` or `AttachToPost`. Anyway, we can add a different task for that.